sync/api/projects.py

```diff
@@ -396,3 +405,79 @@ def get_project_submission(project_id: str, submission_id: str) -> Response[dict
+
+    :param project_id: project ID
+    :type project_id: str
+    :return: Project Configuration Recommendation object
+    :rtype: AWSProjectConfiguration or AzureProjectConfiguration
+    """
+    latest_recommendation = get_default_client().get_latest_project_recommendation(project_id)
+    if latest_recommendation.get("result"):
+        return Response(
+            result=latest_recommendation["result"][0]["recommendation"]["configuration"]
+        )
+
+
+def get_cluster_definition_and_recommendation(
+    project_id: str, cluster_spec_str: str
+) -> Response[dict]:
+    """Print Current Cluster Definition and Project Configuration Recommendatio.
+    Throws error if no cluster recommendation found for project
+
+    :param project_id: project ID
+    :type project_id: str
+    :param cluster_spec_str: Current Cluster Recommendation
+    :type cluster_spec_str: str
+    :return: Current Cluster Definition and Project Configuration Recommendation object
+    :rtype: dict
+    """
+    recommendation_response = get_latest_project_config_recommendation(project_id)
+    if not recommendation_response:
+        logger.info(f"No cluster recommendation found for {project_id}")
+        return Response(error=RecommendationError(message="Recommendation failed"))
+    response_str = json.dumps(recommendation_response.result)
+    return Response(
+        result={
+            "cluster_recommendations": json.loads(response_str),
+            "cluster_definition": json.loads(cluster_spec_str),
+        }
+    )
+
+
+def get_updated_cluster_defintion(
+    project_id: str, cluster_spec_str: str
+) -> Response[Union[AWSProjectConfiguration, AzureProjectConfiguration]]:
+    """Print Cluster Definition merged with Project Configuration Recommendations.
+
+    :param project_id: project ID
+    :type project_id: str
+    :param cluster_spec_str: Current Cluster Recommendation
+    :type cluster_spec_str: str
+    :return: Updated Cluster Definition with Project Configuration Recommendations
+    :rtype: AWSProjectConfiguration or AzureProjectConfiguration
+    """
+    rec_response = get_latest_project_config_recommendation(project_id)
+    if not rec_response.error:
+        # Convert Response result object to str
+        latest_rec_str = json.dumps(rec_response.result)
+        # Convert json string to json
+        latest_recommendation = json.loads(latest_rec_str)
+        cluster_definition = json.loads(cluster_spec_str)
+        for key in latest_recommendation.keys():
+            cluster_definition[key] = latest_recommendation[key]
+
+        # instance_source and driver_instance_source are not
+        # included in recommendation and need to be updated as well
+        driver_recommendation = cluster_definition["node_type_id"]
+        cluster_definition["instance_source"] = {"node_type_id": driver_recommendation}
+        cluster_definition["driver_instance_source"] = {"node_type_id": driver_recommendation}
```

Comment:
We should do a deep merge here like we do when we apply a recommendation from the library:
https://github.com/synccomputingcode/syncsparkpy/blob/dc7a217d4131fcd91ec1f9d3982d9e5c9ed1cbeb/sync/_databricks.py#L633

Reply:
it might be good to review this pathway before merging this:

https://github.com/synccomputingcode/syncsparkpy/blob/dc7a217d4131fcd91ec1f9d3982d9e5c9ed1cbeb/sync/_databricks.py#L524

I think we are rewriting logic that exists already